### PR TITLE
7020 Zynq boards support

### PR DESCRIPTION
**Pull request by acomodi**
This PR is to add support for some of the boards mounting a `Zynq-7Z020`.

To correctly build, the db must be updated with the changes generated with https://github.com/SymbiFlow/prjxray/pull/1162. The PR in prjxray still needs adjustments to be merged, as one (or more) of the fuzzers for the 7020 part end up in a possible infinite-loop, preventing CI to complete.

Once the new db is ready, this PR should correctly turn green.

Boards added with this PR:

- Mars ZX3
- Pynq-Z1
- MicroZed
- ZedBoard

### CODE REVIEW

File: xc7/make/arch_define.cmake [hunk 1]
```diff
@@ -21,7 +21,7 @@ function(ADD_XC7_ARCH_DEFINE)
     DEVICE_FULL_TEMPLATE \${DEVICE}-\${PACKAGE}
     CELLS_SIM ${YOSYS_DATADIR}/xilinx/cells_sim.v ${symbiflow-arch-defs_SOURCE_DIR}/xc7/techmap/cells_sim.v
     VPR_ARCH_ARGS "\
-      --clock_modeling route \
+      --clock_modeling dedicated_network \
```
Comment (litghost) [inline xc7/make/arch_define.cmake]: I don't believe this change is suitable at this time?

Comment (HackerFoo) [reply xc7/make/arch_define.cmake]: I don't think this will work without a clock network model in arch.xml.

**Comment by tmichalak** [comment]
Cleaned up and rebase this PR, but it depends on the [database for Zynq 7020](https://github.com/SymbiFlow/prjxray/pull/1162) and [one related issue](https://github.com/SymbiFlow/prjxray/pull/1185)

File: xc7/make/arch_define.cmake (hunk 1 above)
Comment (tmichalak) [reply xc7/make/arch_define.cmake]: Restored the original version.

**Pull request closed without merging**